backend/src/bork-protocol/utils/token-monitor/filter-tokens-by-interest.ts

```diff
@@ -0,0 +1,225 @@
+import type { TweetQueueService } from '@/services/twitter/analysis-queue.service';
+import type { TwitterConfigService } from '@/services/twitter/twitter-config-service';
+import type { TwitterService } from '@/services/twitter/twitter-service';
+import type {
+  EnrichedToken,
+  InterestingToken,
+  TokenProfile,
+} from '@/types/token-monitor/token';
+import { elizaLogger } from '@elizaos/core';
+import { searchTokenTweets } from './token-search';
+
+/**
+ * Clears old interesting tokens to prevent stale data
+ * @param interestingTokens Map of interesting tokens
+ * @param maxAgeHours Maximum age of tokens to keep (in hours)
+ * @returns Updated map of interesting tokens
+ */
+export function clearOldInterestingTokens(
+  interestingTokens: Map<string, InterestingToken>,
+  maxAgeHours = 24,
+): Map<string, InterestingToken> {
+  const now = new Date();
+  const maxAgeMs = maxAgeHours * 60 * 60 * 1000;
+  const updatedTokens = new Map(interestingTokens);
+
+  for (const [address, token] of updatedTokens.entries()) {
+    const tokenAge = now.getTime() - token.detectedAt.getTime();
+    if (tokenAge > maxAgeMs) {
+      updatedTokens.delete(address);
+    }
+  }
+
+  return updatedTokens;
+}
+
+/**
+ * Identifies if a token is interesting based on various metrics
+ * @param token Enriched token to evaluate
+ * @param interestingTokens Current map of interesting tokens (will be updated)
+ * @returns Token with interest information if interesting, null otherwise
+ */
+export function identifyIfInteresting(
+  token: EnrichedToken,
+  interestingTokens: Map<string, InterestingToken>,
+): InterestingToken | null {
+  try {
+    // Skip tokens without necessary metrics
+    if (!token.metrics || !token.tokenAddress) {
+      return null;
+    }
+
+    let interestReason = '';
+    let score = 0;
+    const metrics = token.metrics;
+
+    // Check various signals that make a token interesting
+
+    // New token with substantial liquidity
+    if (
+      metrics.liquidityMetrics &&
+      metrics.liquidityMetrics.totalLiquidity > 50000 &&
+      metrics.holderCount > 10
+    ) {
+      interestReason = 'New token with substantial liquidity';
+      score += 40;
+    }
+
+    // High holder count for a new token
+    if (metrics.holderCount > 100) {
+      interestReason = interestReason || 'High holder count for a new token';
```

Comment:
i think you want to `interestReason = `${interestReason} || High holder count for a new token` although you need to check if the string is empty. You could make a helper function to append this or I think there's a javascript function for this.
Otherwise an array of reason and then you join them with `||`